packages/clerk-js/src/ui/utils/useFormControl.ts

```diff
@@ -125,3 +154,74 @@ export const buildRequest = (fieldStates: Array<FormControlStateLike>): Record<s
   });
   return request;
 };
+
+type DebouncedFeedback = {
+  debounced: {
+    errorText: string;
+    warningText: string;
+    isSuccessful: boolean;
+    isFocused: boolean;
+    direction: string;
+  };
+};
+
+type DebouncingOption = {
+  hasLostFocus: boolean;
+  warningText: string | undefined;
+  errorText: string | undefined;
+  enableErrorAfterBlur: boolean | undefined;
+  isSuccessful: boolean;
+  isFocused: boolean;
+  direction: string | undefined;
+};
+export const useFormControlFeedback = (
```

Comment:
❓ Please elaborate on this hook. How is skipBlur used? Is it for keyboard navigation?

Reply:
On a separate note, since you already have a DebouncingOption object, consider adding the `skipblur` and `delayTime` keys to the opts object.

Reply:
So i needed a way to group the logic for when an error or the so called "direction" should actually become visible to the end user. I didn't want to add this logic in the useFormControl, and I didn't want to add this inside the FormControl component either.

Reply:
`skipBlur` is used for the keyboard navigation as you said.
I can ditch the option as param and use the `const { submittedWithEnter } = useFormState();` directly inside the hook. This way the callers of `useFormControlFeedback` will not have to worry about passing it.

Reply:
🔧 @panteliselef Please merge skipBluer and delay into the `DebouncingOption` interface.